// ng-cdbangular/esm2015/lib/components/listgroup/listgroupitem/listgroupitem.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class ListgroupitemComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        // @Input() routerLink: string = '';
        this.href = '';
        this.active = false;
        this.hover = false;
    }
    ngOnInit() { }
    ngAfterViewInit() {
        this.renderer.addClass(this.elRef.nativeElement, 'list-group-item');
        if (this.href === '') {
            this.renderer.addClass(this.li.nativeElement, 'w-100');
        }
        if (this.active === true) {
            this.renderer.addClass(this.elRef.nativeElement, 'active');
            if (this.href !== '') {
                this.renderer.addClass(this.Link.link.nativeElement, 'text-white');
            }
            else {
                this.renderer.addClass(this.li.nativeElement, 'text-white');
            }
        }
        if (this.hover === true) {
            this.renderer.addClass(this.elRef.nativeElement, 'list-group-item-action');
        }
        if (this.color) {
            this.renderer.addClass(this.elRef.nativeElement, this.color);
        }
    }
}
ListgroupitemComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBListGroupItem',
                template: "<ng-container [ngSwitch]=\"href !== ''\">\n\n    <ng-container *ngSwitchCase=true>\n        <CDBLink class=\"CDBListGroupLink {{ class }}\" #Link [to]=\"href\">\n            <ng-container *ngTemplateOutlet=\"content\"></ng-container>\n        </CDBLink>\n    </ng-container>\n\n    <ng-container *ngSwitchCase=false>\n        <li #li class=\"CDBListGroupLi {{ class }}\">\n            <ng-container *ngTemplateOutlet=\"content\"></ng-container>\n        </li>\n    </ng-container>\n\n    <ng-template #content>\n        <ng-content></ng-content>\n    </ng-template>\n\n</ng-container>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}.CDBListGroupLi{list-style:none!important}.CDBListGroupLi .nav-link{padding:0}.CDBListGroupLi .nav-link:hover,.CDBListGroupLink{color:#212529}.CDBListGroupLink .nav-link{padding:0}.CDBListGroupLink .nav-link:hover{color:#212529}"]
            },] }
];
ListgroupitemComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
ListgroupitemComponent.propDecorators = {
    href: [{ type: Input }],
    class: [{ type: Input }],
    color: [{ type: Input }],
    active: [{ type: Input }],
    hover: [{ type: Input }],
    li: [{ type: ViewChild, args: ['li', { static: false },] }],
    Link: [{ type: ViewChild, args: ['Link', { static: false },] }]
};
