// ng-cdbangular/esm2015/lib/components/card/card-image/card-image.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class CardImageComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.style = '';
        this.cascade = false;
        this.hover = false;
        this.overlay = 'white-slight';
        this.waves = true;
        this.zoom = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.top === true) {
            this.renderer.addClass(this.img.nativeElement, 'card-img-top');
        }
    }
}
CardImageComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBCardImage',
                template: "<CDBView [zoom]=\"zoom\" [hover]=\"hover\" [cascade]=\"cascade\">\n    <div style=\"touch-action: unset\">\n        <img #img class=\"{{ class }}\" [src]=\"src\" [style]=\"style\">\n    </div>\n</CDBView>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
CardImageComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
CardImageComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    cascade: [{ type: Input }],
    hover: [{ type: Input }],
    overlay: [{ type: Input }],
    src: [{ type: Input }],
    top: [{ type: Input }],
    waves: [{ type: Input }],
    zoom: [{ type: Input }],
    img: [{ type: ViewChild, args: ['img', { static: false },] }]
};
