// ng-cdbangular/esm2015/lib/components/carousel/carousel-inner/carousel-inner.component.js
import { Component, ContentChildren, Input, ViewEncapsulation, Renderer2, ElementRef, } from '@angular/core';
import { CarouselItemComponent } from '../carousel-item/carousel-item.component';
export class CarouselInnerComponent {
    constructor(renderer, elRef) {
        this.renderer = renderer;
        this.elRef = elRef;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
}
CarouselInnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBCarousel-inner',
                template: "<div data-test=\"carousel-inner\" class=\"carousel-inner {{class}}\" [class.active]='active' [style]='style'>\n    <ng-content></ng-content>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
CarouselInnerComponent.ctorParameters = () => [
    { type: Renderer2 },
    { type: ElementRef }
];
CarouselInnerComponent.propDecorators = {
    carouselItems: [{ type: ContentChildren, args: [CarouselItemComponent,] }],
    active: [{ type: Input }],
    class: [{ type: Input }],
    style: [{ type: Input }]
};
