// ng-cdbangular/esm2015/lib/components/navbar/nav-brand/nav-brand.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class NavBrandComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.href) {
            // this.navBrand1.class = 'navbar-brand';
            this.renderer.addClass(this.navBrand1.link.nativeElement, 'navbar-brand');
        }
        else {
            this.renderer.addClass(this.navBrand2.nativeElement, 'navbar-brand');
        }
    }
}
NavBrandComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBNavBrand',
                template: "<ng-container [ngSwitch]=\"href !== undefined\">\n    <ng-container *ngSwitchCase=\"true\">\n        <CDBLink #navBrand1 [to]=\"href\" [class]=\"class\" [style]=\"style\">\n            <ng-container *ngTemplateOutlet=\"content\"></ng-container>\n        </CDBLink>\n    </ng-container>\n\n    <ng-container *ngSwitchCase=\"false\">\n        <div #navBrand2 [class]=\"class\" [style]=\"style\">\n            <ng-container *ngTemplateOutlet=\"content\"></ng-container>\n        </div>\n    </ng-container>\n</ng-container>\n\n<ng-template #content>\n    <ng-content></ng-content>\n</ng-template>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
NavBrandComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NavBrandComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    href: [{ type: Input }],
    navBrand1: [{ type: ViewChild, args: ['navBrand1', { static: false },] }],
    navBrand2: [{ type: ViewChild, args: ['navBrand2', { static: false },] }]
};
