// ng-cdbangular/esm2015/lib/components/modal/modal-body/modal-body.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class ModalBodyComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() { }
}
ModalBodyComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBModalBody',
                template: "<div #modalBody class=\"modal-body {{ class }}\" [style]=\"style\">\n    <ng-content></ng-content>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
ModalBodyComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
ModalBodyComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    modalBody: [{ type: ViewChild, args: ['modalBody', { static: false },] }]
};
