// ng-cdbangular/esm2015/lib/components/navbar/navbar-nav/navbar-nav.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class NavbarNavComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.left = false;
        this.right = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.left === true) {
            this.renderer.addClass(this.elRef.nativeElement, 'mr-auto');
        }
        else {
            this.renderer.addClass(this.elRef.nativeElement, 'justify-content-around');
            this.renderer.addClass(this.navbarNav.nativeElement, 'w-100');
        }
        if (this.right === true) {
            this.renderer.addClass(this.elRef.nativeElement, 'ml-auto');
        }
    }
}
NavbarNavComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBNavbarNav',
                template: "<ul #navbarNav class=\"navbar-nav {{ class }}\" [style]=\"style\">\n    <ng-content></ng-content>\n</ul>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
NavbarNavComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NavbarNavComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    left: [{ type: Input }],
    right: [{ type: Input }],
    navbarNav: [{ type: ViewChild, args: ['navbarNav', { static: false },] }]
};
