// ng-cdbangular/esm2015/lib/components/box/box.component.js
import { Component, Input, ViewChild, ElementRef, Renderer2, ViewEncapsulation, } from '@angular/core';
export class BoxComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.tag = 'div';
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.display) {
            this.renderer.addClass(this.box.nativeElement, `d-${this.display}`);
        }
        if (this.justifyContent) {
            this.renderer.addClass(this.box.nativeElement, `justify-content-${this.justifyContent}`);
        }
        if (this.alignItems) {
            this.renderer.addClass(this.box.nativeElement, `align-items-${this.alignItems}`);
        }
        if (this.alignContent) {
            this.renderer.addClass(this.box.nativeElement, `align-content-${this.alignContent}`);
        }
        if (this.alignSelf) {
            this.renderer.addClass(this.box.nativeElement, `align-self-${this.alignSelf}`);
        }
        if (this.flex) {
            this.renderer.addClass(this.box.nativeElement, `flex-${this.flex}`);
        }
        if (this.color) {
            this.renderer.addClass(this.box.nativeElement, `text-${this.color}`);
        }
        if (this.bg) {
            this.renderer.addClass(this.box.nativeElement, `${this.bg}`);
        }
    }
}
BoxComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBBox',
                template: "<ng-container [ngSwitch]=\"tag\">\n    <span #box class=\"{{ class }}\" [style]=\"style\" *ngSwitchCase=\"'span'\">\n        <ng-container *ngTemplateOutlet=\"content\"></ng-container>\n    </span>\n    <div #box class=\"{{ class }}\" [style]=\"style\" *ngSwitchCase=\"'div'\">\n        <ng-container *ngTemplateOutlet=\"content\"></ng-container>\n    </div>\n    <p #box class=\"{{ class }}\" [style]=\"style\" *ngSwitchCase=\"'p'\">\n        <ng-container *ngTemplateOutlet=\"content\"></ng-container>\n    </p>\n    <section #box class=\"{{ class }}\" [style]=\"style\" *ngSwitchCase=\"'section'\">\n        <ng-container *ngTemplateOutlet=\"content\"></ng-container>\n    </section>\n</ng-container>\n\n<ng-template #content>\n    <ng-content></ng-content>\n</ng-template>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}"]
            },] }
];
BoxComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
BoxComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    alignContent: [{ type: Input }],
    alignItems: [{ type: Input }],
    alignSelf: [{ type: Input }],
    bg: [{ type: Input }],
    color: [{ type: Input }],
    display: [{ type: Input }],
    flex: [{ type: Input }],
    justifyContent: [{ type: Input }],
    m: [{ type: Input }],
    mb: [{ type: Input }],
    ml: [{ type: Input }],
    mr: [{ type: Input }],
    mt: [{ type: Input }],
    mx: [{ type: Input }],
    my: [{ type: Input }],
    p: [{ type: Input }],
    pb: [{ type: Input }],
    pl: [{ type: Input }],
    pr: [{ type: Input }],
    pt: [{ type: Input }],
    px: [{ type: Input }],
    py: [{ type: Input }],
    tag: [{ type: Input }],
    box: [{ type: ViewChild, args: ['box',] }]
};
