// ng-cdbangular/esm2015/lib/components/progress/progress.component.js
import { Component, ElementRef, Input, ViewChild, Renderer2, } from '@angular/core';
export class ProgressComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.text = '';
        this.color = 'dark';
        this.height = 10;
        this.max = 100;
        this.min = 0;
        this.value = 0;
    }
    ngOnInit() { }
    ngAfterViewInit() {
        this.percent = ((this.value - this.min) / (this.max - this.min)) * 100;
        this.renderer.setStyle(this.progress.nativeElement, 'height', `${this.height}px`);
        this.renderer.setStyle(this.progressBar.nativeElement, 'width', `${this.percent}%`);
        this.renderer.setStyle(this.progressBar.nativeElement, 'height', `${this.height}px`);
        // this.renderer.addClass(this.elRef.nativeElement, 'CDBProgress-container');
        this.renderer.setStyle(this.elRef.nativeElement, 'display', 'block');
        this.renderer.setStyle(this.elRef.nativeElement, 'width', '100%');
        if (this.class) {
            let classes = this.class.split(' ');
            classes.forEach((className) => {
                this.renderer.addClass(this.progress.nativeElement, className);
            });
        }
        if (this.barClass) {
            let classes = this.class.split(' ');
            classes.forEach((className) => {
                this.renderer.addClass(this.progressBar.nativeElement, className);
            });
        }
        if (this.color === 'primary') {
            this.renderer.addClass(this.progress.nativeElement, 'CDBProgress-primary');
            this.renderer.addClass(this.progressBar.nativeElement, this.color);
            this.renderer.addClass(this.progressText.nativeElement, 'progress-text-primary');
        }
        if (this.color === 'secondary') {
            this.renderer.addClass(this.progress.nativeElement, 'CDBProgress-secondary');
            this.renderer.addClass(this.progressBar.nativeElement, this.color);
            this.renderer.addClass(this.progressText.nativeElement, 'progress-text-secondary');
        }
        if (this.color === 'success') {
            this.renderer.addClass(this.progress.nativeElement, 'CDBProgress-success');
            this.renderer.addClass(this.progressBar.nativeElement, this.color);
            this.renderer.addClass(this.progressText.nativeElement, 'progress-text-success');
        }
        if (this.color === 'danger') {
            this.renderer.addClass(this.progress.nativeElement, 'CDBProgress-danger');
            this.renderer.addClass(this.progressBar.nativeElement, this.color);
            this.renderer.addClass(this.progressText.nativeElement, 'progress-text-danger');
        }
        if (this.color === 'warning') {
            this.renderer.addClass(this.progress.nativeElement, 'CDBProgress-warning');
            this.renderer.addClass(this.progressBar.nativeElement, this.color);
            this.renderer.addClass(this.progressText.nativeElement, 'progress-text-warning');
        }
        if (this.color === 'info') {
            this.renderer.addClass(this.progress.nativeElement, 'CDBProgress-info');
            this.renderer.addClass(this.progressBar.nativeElement, this.color);
            this.renderer.addClass(this.progressText.nativeElement, 'progress-text-info');
        }
        if (this.color === 'dark') {
            this.renderer.addClass(this.progressBar.nativeElement, this.color);
        }
    }
}
ProgressComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBProgress',
                template: "<div #progressText class=\"progress-text\">{{ text }}</div>\n<div #progress class=\"progress CDBProgress-default\">\n    <div #progressBar class=\"progress-bar\" role=\"progressbar\">\n    </div>\n</div>",
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}.CDBProgress-default{background:rgba(0,0,0,.1568627450980392);border-radius:20px;width:100%}.progress-bar{border-radius:20px}.CDBProgress-primary{background:rgba(62,152,199,.1568627450980392)!important}.CDBProgress-secondary{background:rgba(115,86,191,.1568627450980392)!important}.CDBProgress-danger{background:rgba(225,25,0,.1568627450980392)!important}.CDBProgress-warning{background:rgba(102,81,44,.1568627450980392)!important}.CDBProgress-info{background:rgba(23,162,184,.1568627450980392)!important}.CDBProgress-success{background:rgba(5,148,79,.1568627450980392)!important}.progress-text{color:#000;display:flex;justify-content:center;width:100%}.progress-text-primary{color:#3e98c7!important}.progress-text-secondary{color:#867d9c!important}.progress-text-danger{color:#e11900!important}.progress-text-warning{color:#66512c!important}.progress-text-info{color:#17a2b8!important}.progress-text-success{color:#05944f!important}"]
            },] }
];
ProgressComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
ProgressComponent.propDecorators = {
    barClass: [{ type: Input }],
    class: [{ type: Input }],
    text: [{ type: Input }],
    color: [{ type: Input }],
    height: [{ type: Input }],
    max: [{ type: Input }],
    min: [{ type: Input }],
    value: [{ type: Input }],
    progressText: [{ type: ViewChild, args: ['progressText',] }],
    progress: [{ type: ViewChild, args: ['progress',] }],
    progressBar: [{ type: ViewChild, args: ['progressBar',] }]
};
