// ng-cdbangular/esm2015/lib/components/navbar/nav-toggle/nav-toggle.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class NavToggleComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.type = 'button';
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
        if (this.image) {
            this.imagePath = `background-image: url(${this.image})`;
        }
    }
    ngAfterViewInit() {
        if (this.left === true) {
            this.renderer.addClass(this.navToggle.nativeElement, 'navbar-toggler-left');
        }
        if (this.right === true) {
            this.renderer.addClass(this.navToggle.nativeElement, 'navbar-toggler-right');
        }
    }
}
NavToggleComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBNavToggle',
                template: "<ng-container *ngIf=\"image; else elseTemplate\">\n    <button #navToggle [type]=\"type\" class=\"navbar-toggler {{ class }}\" [style]=\"style\">\n        <span class=\"navbar-toggler-icon\" [style]=\"imagePath\"></span>\n    </button>\n</ng-container>\n\n<ng-template #elseTemplate>\n    <button #navToggle [type]=\"type\" class=\"navbar-toggler {{ class }}\" [style]=\"style\">\n        <span class=\"navbar-toggler-icon\"></span>\n    </button>\n</ng-template>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
NavToggleComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NavToggleComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    image: [{ type: Input }],
    left: [{ type: Input }],
    right: [{ type: Input }],
    type: [{ type: Input }],
    navToggle: [{ type: ViewChild, args: ['navToggle', { static: false },] }]
};
