// ng-cdbangular/esm2015/lib/components/carousel/carousel-control/carousel-control.component.js
import { Component, Output, EventEmitter, Input, ViewChild, ElementRef, Renderer2, ViewEncapsulation, } from '@angular/core';
export class CarouselControlComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.onClick = new EventEmitter();
    }
    ngOnInit() {
        if (this.direction === 'prev') {
            this.text = 'Previous';
        }
        else if (this.direction === 'next') {
            this.text = 'Next';
        }
    }
    ngAfterViewInit() {
        if (this.testimonial) {
            const arrow = this.direction === 'prev' ? 'left' : 'right';
            this.renderer.addClass(this.carouselcontrol.nativeElement, `carousel-control-${this.direction}`);
            this.renderer.addClass(this.carouselcontrol.nativeElement, arrow);
            this.renderer.addClass(this.carouselcontrol.nativeElement, `carousel-control`);
            this.renderer.addClass(this.caret.nativeElement, `icon-${this.direction}`);
        }
        else {
            this.renderer.addClass(this.carouselcontrol.nativeElement, `carousel-control-${this.direction}`);
            this.renderer.addClass(this.caret.nativeElement, `carousel-control-${this.direction}-icon`);
        }
        if (this.multiItem) {
            this.renderer.addClass(this.carouselcontrol.nativeElement, `btn-floating`);
        }
    }
}
CarouselControlComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBCarousel-control',
                template: "<a #carouselcontrol (click)=\"onClick.emit()\">\n    <div>\n        <span #caret [class]='caretClasses' aria-hidden=\"true\"></span>\n        <span class=\"sr-only\">{{text}}</span>\n    </div>\n</a>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
CarouselControlComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
CarouselControlComponent.propDecorators = {
    class: [{ type: Input }],
    direction: [{ type: Input }],
    multiItem: [{ type: Input }],
    testimonial: [{ type: Input }],
    caretClasses: [{ type: Input }],
    carouselcontrol: [{ type: ViewChild, args: ['carouselcontrol', { static: false },] }],
    caret: [{ type: ViewChild, args: ['caret', { static: false },] }],
    onClick: [{ type: Output }]
};
