// ng-cdbangular/esm2015/lib/components/navbar/navbar.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class NavbarComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.isCollapsed = false;
        this.expand = false;
        this.scrolling = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
        if (this.scrolling === true || this.scrollingNavbarOffset) {
            window.addEventListener('scroll', this.handleScroll);
        }
        // if (this.scrolling === true || this.scrollingNavbarOffset) {
        //   window.removeEventListener('scroll', this.handleScroll);
        // }
    }
    ngAfterViewInit() {
        if (this.color) {
            this.renderer.addClass(this.nav.nativeElement, this.color);
        }
        if (this.light === true) {
            this.renderer.addClass(this.nav.nativeElement, 'navbar-light');
        }
        if (this.dark === true) {
            this.renderer.addClass(this.nav.nativeElement, 'navbar-dark');
        }
        if (this.sticky) {
            this.renderer.addClass(this.nav.nativeElement, `sticky-${this.sticky}`);
        }
        if (this.fixed) {
            this.renderer.addClass(this.nav.nativeElement, `fixed-${this.fixed}`);
        }
        if (this.scrolling === true || this.scrollingNavbarOffset) {
            this.renderer.addClass(this.nav.nativeElement, 'scrolling-navbar');
        }
        if (this.double === true) {
            this.renderer.addClass(this.nav.nativeElement, 'double-nav');
        }
        if (this.isCollapsed === true) {
            this.renderer.addClass(this.nav.nativeElement, 'top-nav-collapse');
        }
        if (this.expand === true || this.expand === 'xs') {
            this.renderer.addClass(this.nav.nativeElement, 'navbar-expand');
        }
        else if (this.expand === 'sm' ||
            this.expand === 'md' ||
            this.expand === 'lg' ||
            this.expand === 'xl') {
            this.renderer.addClass(this.nav.nativeElement, `navbar-expand-${this.expand}`);
        }
    }
    handleScroll() {
        const scrollingNavbarOffset = this.scrollingNavbarOffset || 50;
        if (window.pageYOffset > scrollingNavbarOffset) {
            this.isCollapsed = true;
        }
        else {
            this.isCollapsed = false;
        }
    }
}
NavbarComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBNavbar',
                template: "<nav #nav class=\"navbar {{ class }}\" [style]=\"style\" role='navigation'>\n    <ng-content></ng-content>\n</nav>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}"]
            },] }
];
NavbarComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NavbarComponent.propDecorators = {
    isCollapsed: [{ type: Input }],
    expand: [{ type: Input }],
    light: [{ type: Input }],
    dark: [{ type: Input }],
    sticky: [{ type: Input }],
    fixed: [{ type: Input }],
    scrolling: [{ type: Input }],
    color: [{ type: Input }],
    class: [{ type: Input }],
    style: [{ type: Input }],
    scrollingNavbarOffset: [{ type: Input }],
    double: [{ type: Input }],
    transparent: [{ type: Input }],
    nav: [{ type: ViewChild, args: ['nav', { static: false },] }]
};
