// ng-cdbangular/esm2015/lib/components/pagination/pagelink/pagelink.component.js
import { Component, Input, ViewEncapsulation, ViewChild, ElementRef, Renderer2, } from '@angular/core';
export class PagelinkComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.disabled = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
}
PagelinkComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBPagelink',
                template: "<a #pagelink [style]='style' class=\"page-link {{class}}\" [class.disabled]=\"disabled\">\n    <ng-content></ng-content>\n</a>",
                encapsulation: ViewEncapsulation.None,
                styles: [".page-link{cursor:pointer;margin:0 7.5px}.page-link.disabled{background:#ccc;color:#555;pointer-events:none}"]
            },] }
];
PagelinkComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
PagelinkComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    string: [{ type: Input }],
    disabled: [{ type: Input }],
    pagelink: [{ type: ViewChild, args: ['pagelink', { static: false },] }]
};
