// ng-cdbangular/esm2015/lib/components/dropdown/dropdown.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class DropdownComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() { }
}
DropdownComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBDropDown',
                template: "<span #dropdown role=\"dropdown\" class=\"CDBDropDown dropdown {{ class }}\" [style]=\"style\">\n    <ng-content></ng-content>\n</span>",
                encapsulation: ViewEncapsulation.None,
                styles: [".CDBDropDown{cursor:pointer;display:flex;flex-direction:column;position:relative}"]
            },] }
];
DropdownComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
DropdownComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    dropdown: [{ type: ViewChild, args: ['dropdown', { static: false },] }]
};
