// ng-cdbangular/esm2015/lib/components/sidebar/sidebar-header/sidebar-header.component.js
import { SidebarService } from './../sidebar.service';
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class SidebarHeaderComponent {
    constructor(elRef, renderer, sidebarService) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.sidebarService = sidebarService;
        this.toggled = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        this.dynamicallyHandleToggleSidebar();
        window.addEventListener('resize', () => {
            this.dynamicallyHandleToggleSidebar();
        });
    }
    toggleSidebar() {
        this.toggled = !this.toggled;
        if (this.toggled === true) {
            this.renderer.addClass(this.sidebarHeader.nativeElement, 'toggled');
        }
        else {
            if (this.sidebarHeader.nativeElement.classList.contains('toggled')) {
                this.renderer.removeClass(this.sidebarHeader.nativeElement, 'toggled');
            }
        }
        this.sidebarService.sidebarState.next(this.toggled);
    }
    dynamicallyHandleToggleSidebar() {
        if (matchMedia('(max-width:991.98px)').matches) {
            this.toggled = true;
            this.renderer.addClass(this.sidebarHeader.nativeElement, 'toggled');
        }
        else {
            this.toggled = false;
            if (this.sidebarHeader.nativeElement.classList.contains('toggled')) {
                this.renderer.removeClass(this.sidebarHeader.nativeElement, 'toggled');
            }
        }
        this.sidebarService.sidebarState.next(this.toggled);
    }
}
SidebarHeaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBSidebarHeader',
                template: "<div class=\"CDBSidebarHeader-default\">\n    <div #sidebarHeader class=\"head-div {{ class }}\" [style]=\"style\">\n        <span *ngIf=\"prefix\" class=\"icon-suffix\" (click)=\"toggleSidebar()\">\n            <ng-container *ngTemplateOutlet=\"prefix\"></ng-container>\n        </span>\n        <span class=\"head-text\">\n            <ng-content></ng-content>\n        </span>\n    </div>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}.CDBSidebarHeader-default{border-bottom:1px solid hsla(0,0%,67.8%,.2)}.CDBSidebarHeader-default .head-div{font-size:17px;font-weight:700;letter-spacing:1px;overflow:hidden;padding:24px 29px;white-space:nowrap}.CDBSidebarHeader-default .head-div .icon-suffix{cursor:pointer;margin:0 23px}.CDBSidebarHeader-default .head-div.toggled .head-text{display:none}.CDBSidebarHeader-default .head-div.toggled .icon-suffix{margin:0}"]
            },] }
];
SidebarHeaderComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 },
    { type: SidebarService }
];
SidebarHeaderComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    prefix: [{ type: Input }],
    sidebarHeader: [{ type: ViewChild, args: ['sidebarHeader', { static: false },] }]
};
