// ng-cdbangular/esm2015/lib/components/view/view.component.js
import { Component, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class ViewComponent {
    constructor(renderer) {
        this.renderer = renderer;
        this.hover = false;
        this.rounded = false;
        this.waves = false;
        this.zoom = false;
        this.cascade = false;
        this.src = '';
        this.style = '';
    }
    ngOnInit() {
        if (this.src !== '') {
            const BA = this.fixed === true ? 'background-attachment: fixed;' : '';
            const BI = `background-image: url("${this.src}");`;
            const BP = 'background-position: center center;';
            const BR = 'background-repeat: no-repeat;';
            const BS = 'background-size: cover;';
            const height = 'height: 100vh;';
            this.style = BA + BI + BP + BR + BS + height;
        }
    }
    ngAfterViewInit() {
        if (this.rounded === true) {
            this.renderer.addClass(this.view.nativeElement, 'CDBView-rounded');
        }
        if (this.hover === true) {
            this.renderer.addClass(this.view.nativeElement, 'CDBView-hover');
        }
        if (this.waves === true) {
            this.renderer.addClass(this.view.nativeElement, 'Ripple-parent');
        }
        if (this.zoom === true) {
            this.renderer.addClass(this.view.nativeElement, 'CDBView-zoom');
        }
    }
}
ViewComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBView',
                template: "<div #view class=\"CDBView-default\" [style]=\"style\">\n    <ng-content></ng-content>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}.CDBView-default{-moz-border-radius:0;-ms-border-radius:0;-o-border-radius:0;-webkit-border-radius:0;border-radius:0;cursor:default;overflow:hidden;position:relative}.CDBView-zoom img,.CDBView-zoom video{transition:all .2s linear}.CDBView-zoom:hover img,.CDBView-zoom:hover video{transform:scale(1.1)}.mask{background-attachment:fixed;bottom:0;height:100%;left:0;overflow:hidden;position:absolute;right:0;top:0;width:100%}img,video{display:block;position:relative}.video.video-intro{height:auto;left:50%;min-height:100%;min-width:100%;top:50%;transform:translateX(-50%) translateY(-50%);transition:opacity 1s;width:auto;z-index:-100}.CDBView-rounded{-moz-border-radius:.25rem!important;-ms-border-radius:.25rem!important;-o-border-radius:.25rem!important;-webkit-border-radius:.25rem!important;border-radius:.25rem!important}.CDBView-hover .mask{opacity:0;transition:all .4s ease-in-out}.CDBView-hover .mask:hover{opacity:1}"]
            },] }
];
ViewComponent.ctorParameters = () => [
    { type: Renderer2 }
];
ViewComponent.propDecorators = {
    class: [{ type: Input }],
    fixed: [{ type: Input }],
    hover: [{ type: Input }],
    rounded: [{ type: Input }],
    waves: [{ type: Input }],
    zoom: [{ type: Input }],
    cascade: [{ type: Input }],
    src: [{ type: Input }],
    view: [{ type: ViewChild, args: ['view', { static: false },] }]
};
