// ng-cdbangular/esm2015/lib/components/link/link.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class LinkComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.active = false;
        this.disabled = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.disabled === true) {
            this.renderer.addClass(this.link.nativeElement, 'disabled');
        }
        if (this.active === true) {
            this.renderer.addClass(this.link.nativeElement, 'active');
        }
    }
}
LinkComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBLink',
                template: "<a #link [routerLink]=\"to\" class=\"nav-link {{ class }}\" [style]=\"style\">\n    <ng-content></ng-content>\n</a>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
LinkComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
LinkComponent.propDecorators = {
    active: [{ type: Input }],
    disabled: [{ type: Input }],
    class: [{ type: Input }],
    style: [{ type: Input }],
    to: [{ type: Input }],
    link: [{ type: ViewChild, args: ['link', { static: false },] }]
};
