// ng-cdbangular/esm2015/lib/components/table/table-pagination/table-pagination.component.js
import { Component, Input, ViewEncapsulation } from '@angular/core';
export class TablePaginationComponent {
    constructor() { }
    numberOfPages() {
        return Math.ceil(this.tableEl.getDataSource().length / this.tableEl.getEntries());
    }
    pagesArray() {
        let pages = [];
        for (let i = 0; i < this.numberOfPages(); i++) {
            pages.push(i);
        }
        return pages;
    }
    changeActivePage(page) {
        if (!(page < 0 || page > this.pagesArray().length - 1)) {
            this.tableEl.changeActivePage(page);
        }
    }
    ngOnInit() { }
}
TablePaginationComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBTable-pagination',
                template: "<div data-test='datatable-pagination' class='col-sm-12 col-md-12 pr-0'>\n    <div>\n        <CDBPagination [sm]='true'>\n            <CDBPagelink (click)='changeActivePage(tableEl.activePage-1)' [disabled]='tableEl.activePage <= 0'>\n                <span>Prev</span>\n            </CDBPagelink>\n            <CDBPageitem *ngFor='let page of pagesArray()' (click)='changeActivePage(page)'>\n                {{page + 1}}\n            </CDBPageitem>\n            <CDBPagelink aria-label='label' (click)='changeActivePage(tableEl.activePage+1)'\n                [disabled]='tableEl.activePage == numberOfPages() - 1'>\n                <span>Next</span>\n            </CDBPagelink>\n        </CDBPagination>\n    </div>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
TablePaginationComponent.ctorParameters = () => [];
TablePaginationComponent.propDecorators = {
    tableEl: [{ type: Input }]
};
