// ng-cdbangular/esm2015/lib/components/select/select.component.js
import { Component, Input, Renderer2, ElementRef } from '@angular/core';
export class SelectComponent {
    constructor(renderer, elRef) {
        this.renderer = renderer;
        this.elRef = elRef;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
}
SelectComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBSelect',
                template: "<select class=\"custom-select {{class}}\" [style]=\"style\">\n    <option value=\"\">{{ selected }}</option>\n\n    <ng-container *ngFor=\"let option of options; let i=index\">\n        <option [class]='optionClass' [value]=\"option.value\">{{ option.text }}</option>\n    </ng-container>\n</select>",
                styles: [""]
            },] }
];
SelectComponent.ctorParameters = () => [
    { type: Renderer2 },
    { type: ElementRef }
];
SelectComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    options: [{ type: Input }],
    selected: [{ type: Input }],
    optionClass: [{ type: Input }]
};
