// ng-cdbangular/esm2015/lib/components/switch/switch.component.js
import { Component, Input, ViewEncapsulation } from '@angular/core';
export class SwitchComponent {
    constructor() {
        this.checked = false;
    }
    ngOnInit() { }
    handleChange() {
        this.checked = !this.checked;
    }
}
SwitchComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBSwitch',
                template: "<div class=\"switch-container\">\n    <label>\n        <input [checked]=checked (click)=\"handleChange()\" class=\"switch\" type=\"checkbox\" />\n        <div>\n            <div></div>\n        </div>\n    </label>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [".switch-container{left:50%;position:relative;top:50%;transform:translate3d(-50%,-50%,0)}input[type=checkbox].switch{opacity:0;position:relative}input[type=checkbox].switch+div{background-color:#00b0f2;border-radius:999px;cursor:pointer;height:20px;transition-duration:.4s;transition-property:background-color,box-shadow;vertical-align:middle;width:40px}input[type=checkbox].switch+div span{color:#fff;font-size:1.6rem;margin-top:12px;position:absolute}input[type=checkbox].switch+div span:first-child{margin-left:15px}input[type=checkbox].switch+div span:nth-child(2){margin-left:55px}input[type=checkbox].switch:checked+div{background-color:#3b89ec;background-position:0 0;width:94px}input[type=checkbox].switch+div{height:48px;width:94px}input[type=checkbox].switch:checked+div{background-color:#57bb00}input[type=checkbox].switch+div>div{background:#fff;border-radius:inherit;float:left;height:46px;margin-left:1px;margin-top:1px;pointer-events:none;transition-duration:.4s;transition-property:transform,background-color;transition-timing-function:cubic-bezier(1,0,0,1);width:46px}input[type=checkbox].switch:checked+div>div{background-color:#fff;transform:translate3d(20px,0,0)}input[type=checkbox].bigswitch.switch+div>div{height:44px;margin-top:1px;width:44px}input[type=checkbox].switch:checked+div>div{transform:translate3d(46px,0,0)}"]
            },] }
];
SwitchComponent.ctorParameters = () => [];
SwitchComponent.propDecorators = {
    checked: [{ type: Input }]
};
