// ng-cdbangular/esm2015/lib/components/button/button.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class ButtonComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.color = 'primary';
        this.disabled = false;
        this.flat = false;
        this.size = 'medium';
        this.type = 'button';
        this.outline = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        this.renderer.addClass(this.button.nativeElement, this.color);
        this.renderer.addClass(this.button.nativeElement, this.size);
        if (this.outline === true) {
            this.renderer.addClass(this.button.nativeElement, 'outline');
        }
        if (this.flat === true) {
            this.renderer.addClass(this.button.nativeElement, 'flat');
        }
        if (this.block == true) {
            this.renderer.setStyle(this.elRef.nativeElement, 'width', '100%');
            this.renderer.addClass(this.button.nativeElement, 'btn-block');
        }
        if (this.size) {
            this.renderer.addClass(this.button.nativeElement, this.size);
        }
    }
}
ButtonComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBBtn',
                template: "<button #button [type]=\"type\" class=\"btn CDBbtn {{ class }}\" [style]=\"style\" [disabled]=\"disabled\">\n    <ng-content></ng-content>\n</button>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}.CDBbtn{border-radius:0;box-shadow:0 2px 5px 0 rgba(0,0,0,.16),0 2px 10px 0 rgba(0,0,0,.12)}.xl{padding:30px}.large,.xl{font-size:1.25rem}.large{padding:20px 30px}.medium{font-size:1rem;padding:10px 20px}.small{font-size:.75rem;padding:5px 10px}.flat{box-shadow:none}.circle{border-radius:30px}.outline.primary{background-color:initial;border-width:2px;color:#276ef1}.outline.primary:hover{background-color:#fdfdfd;color:rgba(39,110,241,.7411764705882353)}.outline.secondary{background-color:initial;border-width:2px;color:#7356bf}.outline.secondary:hover{background-color:#fdfdfd;color:rgba(115,86,191,.7411764705882353)}.outline.success{background-color:initial;border-width:2px;color:#05944f}.outline.success:hover{background-color:#fdfdfd;color:rgba(5,148,79,.7411764705882353)}.outline.danger{background-color:initial;border-width:2px;color:#e11900}.outline.danger:hover{background-color:#fdfdfd;color:rgba(225,25,0,.7411764705882353)}.outline.dark{background-color:initial;border-width:2px;color:#000}.outline.dark:hover{background-color:#fdfdfd;color:rgba(0,0,0,.7411764705882353)}.outline.warning{background-color:initial;border-width:2px;color:#ffe975}.outline.warning:hover{background-color:#fdfdfd;color:rgba(255,233,117,.7411764705882353)}.outline.info{background-color:initial;border-width:2px;color:#17a2b8}.outline.info:hover{background-color:#fdfdfd;color:rgba(23,162,184,.7411764705882353)}"]
            },] }
];
ButtonComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
ButtonComponent.propDecorators = {
    block: [{ type: Input }],
    color: [{ type: Input }],
    disabled: [{ type: Input }],
    flat: [{ type: Input }],
    size: [{ type: Input }],
    type: [{ type: Input }],
    class: [{ type: Input }],
    style: [{ type: Input }],
    outline: [{ type: Input }],
    button: [{ type: ViewChild, args: ['button', { static: false },] }]
};
