// ng-cdbangular/esm2015/lib/components/pagination/pagination.component.js
import { Component, Input, ViewEncapsulation, ElementRef, Renderer2, ContentChildren, } from '@angular/core';
import { PageitemComponent } from './pageitem/pageitem.component';
import { PagelinkComponent } from './pagelink/pagelink.component';
export class PaginationComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.color = 'dark';
        this.circle = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        this.pageitems.forEach((element) => {
            this.renderer.addClass(element.pageitem.nativeElement, this.color);
        });
        this.pagelinks.forEach((element) => {
            this.renderer.addClass(element.pagelink.nativeElement, this.color);
        });
        if (this.sm) {
            this.pageitems.forEach((element) => {
                this.renderer.addClass(element.pageitem.nativeElement, `sm`);
            });
            this.pagelinks.forEach((element) => {
                this.renderer.addClass(element.pagelink.nativeElement, `sm`);
            });
            //  this.renderer.addClass(this.pagelink.nativeElement, `sm`);
        }
        if (this.circle) {
            this.pageitems.forEach((element) => {
                this.renderer.addClass(element.pageitem.nativeElement, `circle`);
            });
            this.pagelinks.forEach((element) => {
                this.renderer.addClass(element.pagelink.nativeElement, `circle`);
            });
        }
        if (this.size === 'small') {
            this.pageitems.forEach((element) => {
                this.renderer.addClass(element.pageitem.nativeElement, `small`);
            });
        }
        if (this.size === 'big') {
            this.pageitems.forEach((element) => {
                this.renderer.addClass(element.pageitem.nativeElement, `big`);
            });
        }
    }
}
PaginationComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBPagination',
                template: "<ul class=\"pagination {{class}}\" [style]='style'>\n    <ng-content></ng-content>\n</ul>",
                encapsulation: ViewEncapsulation.None,
                styles: [".pagination{align-items:center;border:.75px solid #ccc;border-radius:0;padding:15px!important}.pagination .primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.pagination .primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.pagination .secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.pagination .secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.pagination .success{background-color:#05944f;border-color:#05944f;color:#fff}.pagination .success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.pagination .danger{background-color:#e11900;border-color:#e11900;color:#fff}.pagination .danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.pagination .warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.pagination .warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.pagination .dark{background-color:#000;border-color:#000;color:#fff}.pagination .dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.pagination .light{background-color:#eee;border-color:#eee;color:#000}.pagination .light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.pagination .white{background-color:#fff;border-color:#fff;color:#000}.pagination .white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.pagination .info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.pagination .info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}.pagination .page-link{align-items:center;border:.5px solid #ccc;border-radius:0;display:flex;justify-content:center;min-height:50px;min-width:50px}.pagination .page-link.sm{min-height:30px;min-width:30px}.pagination .page-link.circle{border-radius:50%}.pagination .page-item{align-items:center;border:.5px solid #ccc;border-radius:0;display:flex;font-size:1rem;justify-content:center;min-height:50px;min-width:50px;text-align:center}.pagination .page-item.sm{min-height:30px;min-width:30px}.pagination .page-item.circle{border-radius:50%}.pagination .page-item.small{font-size:.75rem}.pagination .page-item.big{font-size:1.25rem}"]
            },] }
];
PaginationComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
PaginationComponent.propDecorators = {
    pageitems: [{ type: ContentChildren, args: [PageitemComponent,] }],
    pagelinks: [{ type: ContentChildren, args: [PagelinkComponent,] }],
    class: [{ type: Input }],
    style: [{ type: Input }],
    color: [{ type: Input }],
    shape: [{ type: Input }],
    size: [{ type: Input }],
    sm: [{ type: Input }],
    circle: [{ type: Input }]
};
