// ng-cdbangular/esm2015/lib/components/sidebar/sidebar.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
import { SidebarService } from './sidebar.service';
export class SidebarComponent {
    constructor(elRef, renderer, sidebarService) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.sidebarService = sidebarService;
        this.textColor = '#ffffff';
        this.backgroundColor = '#000000';
        this.toggled = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
        this.sidebarService.sidebarState.subscribe((state) => {
            this.toggled = state;
            this.setToggled();
        });
    }
    ngAfterViewInit() {
        this.sidebarService.textColor.next(this.textColor);
        this.sidebarService.backgroundColor.next(this.backgroundColor);
        this.renderer.setStyle(this.sidebar.nativeElement, 'background-color', this.backgroundColor);
        this.renderer.setStyle(this.sidebar.nativeElement, 'color', this.textColor);
    }
    setToggled() {
        if (this.toggled === true) {
            this.renderer.addClass(this.sidebar.nativeElement, 'toggled');
        }
        else {
            this.renderer.removeClass(this.sidebar.nativeElement, 'toggled');
        }
    }
}
SidebarComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBSidebar',
                template: "<div #sidebar class=\"CDBSidebar-default {{ class }}\" [style]=\"style\">\n    <div class=\"SidebarInner\">\n        <div class=\"SidebarLayout\">\n            <ng-content></ng-content>\n        </div>\n    </div>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}.CDBSidebar-default{height:100%;min-width:270px;position:relative;text-align:left;transition:width,left,right,.3s;width:270px;z-index:1009}.CDBSidebar-default.toggled{min-width:80px;width:80px}.SidebarInner,.SidebarLayout{height:100%;position:relative;z-index:101}.SidebarLayout{display:flex;flex-direction:column;overflow-x:hidden;overflow-y:auto}"]
            },] }
];
SidebarComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 },
    { type: SidebarService }
];
SidebarComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    textColor: [{ type: Input }],
    backgroundColor: [{ type: Input }],
    sidebar: [{ type: ViewChild, args: ['sidebar', { static: false },] }]
};
