// ng-cdbangular/esm2015/lib/components/navbar/nav-link/nav-link.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class NavLinkComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.style = '';
        this.active = false;
        this.disabled = false;
        this.link = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.class) {
            let classes = this.class.split(' ');
            classes.forEach((className) => {
                this.renderer.addClass(this.navLink.link.nativeElement, className);
            });
        }
        if (this.active === true) {
            this.renderer.addClass(this.navLink.link.nativeElement, 'active');
        }
    }
}
NavLinkComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBNavLink',
                template: "<CDBLink #navLink [class]=\"'nav-link'\" [style]=\"style\">\n    <ng-content></ng-content>\n</CDBLink>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
NavLinkComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NavLinkComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    active: [{ type: Input }],
    disabled: [{ type: Input }],
    link: [{ type: Input }],
    to: [{ type: Input }],
    navLink: [{ type: ViewChild, args: ['navLink', { static: false },] }]
};
