// ng-cdbangular/esm2015/lib/components/breadcrumb/breadcrumb.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class BreadcrumbComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.bold === true) {
            this.renderer.addClass(this.ol.nativeElement, 'CDBBreadcrumb-bold');
        }
        if (this.light === true) {
            this.renderer.addClass(this.ol.nativeElement, 'CDBBreadcrumb-light');
        }
        if (this.uppercase === true) {
            this.renderer.addClass(this.ol.nativeElement, 'CDBBreadcrumb-uppercase');
        }
        if (this.circle === true) {
            this.renderer.addClass(this.ol.nativeElement, 'CDBBreadcrumb-circle');
        }
    }
}
BreadcrumbComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBBreadcrumb',
                template: "<nav>\n    <ol #ol class=\"CDBBreadcrumb-default breadcrumb {{ class }}\" [style]=\"style\">\n        <ng-content></ng-content>\n    </ol>\n</nav>",
                encapsulation: ViewEncapsulation.None,
                styles: [".CDBBreadcrumb-default{-moz-border-radius:0;-ms-border-radius:0;-o-border-radius:0;-webkit-border-radius:0;border-radius:0;text-transform:capitalize}.CDBBreadcrumb-default a{display:unset!important}.CDBBreadcrumb-bold{font-weight:700}.CDBBreadcrumb-light{font-weight:light}.CDBBreadcrumb-circle{-moz-border-radius:50px;-ms-border-radius:50px;-o-border-radius:50px;-webkit-border-radius:50px;border-radius:50px}.CDBBreadcrumb-uppercase{text-transform:uppercase}"]
            },] }
];
BreadcrumbComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
BreadcrumbComponent.propDecorators = {
    bold: [{ type: Input }],
    light: [{ type: Input }],
    circle: [{ type: Input }],
    uppercase: [{ type: Input }],
    class: [{ type: Input }],
    style: [{ type: Input }],
    ol: [{ type: ViewChild, args: ['ol', { static: false },] }]
};
