// ng-cdbangular/esm2015/lib/components/table/table.component.js
import { Component, Input } from '@angular/core';
export class TableComponent {
    constructor() { }
    ngOnInit() {
    }
}
TableComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBTable',
                template: "<div\n    data-test=\"table\"\n    [class.table-dark]='dark'\n    [class.table-responsive]='responsive'\n    [class.table-responsive-sm]='responsiveSm'\n    [class.table-responsive-md]='responsiveMd'\n    [class.table-responsive-lg]='responsiveLg'\n    [class.table-responsive-xl]='responsiveXl'\n    [class.table-wrapper-scroll-y]='scrollY'\n>\n    <table\n        class=\"table\"\n        [class.w-auto]='autoWidth'\n        [class.table-bordered]='bordered'\n        [class.table-borderless]='borderless'\n        [class.btn-table]='btn'\n        [class.table-fixed]='fixed'\n        [class.table-hover]='hover'\n        [class.table-striped]='striped'\n        [class.able-sm]='small'\n    >\n        <ng-content></ng-content>\n    </table>\n</div>",
                styles: [""]
            },] }
];
TableComponent.ctorParameters = () => [];
TableComponent.propDecorators = {
    autoWidth: [{ type: Input }],
    bordered: [{ type: Input }],
    borderless: [{ type: Input }],
    btn: [{ type: Input }],
    class: [{ type: Input }],
    dark: [{ type: Input }],
    fixed: [{ type: Input }],
    hover: [{ type: Input }],
    maxHeight: [{ type: Input }],
    responsive: [{ type: Input }],
    responsiveLg: [{ type: Input }],
    responsiveMd: [{ type: Input }],
    responsiveSm: [{ type: Input }],
    responsiveXl: [{ type: Input }],
    scrollY: [{ type: Input }],
    small: [{ type: Input }],
    striped: [{ type: Input }],
    theadColor: [{ type: Input }],
    wrapperClassName: [{ type: Input }]
};
