// ng-cdbangular/esm2015/lib/components/carousel/carousel-item/carousel-item.component.js
import { Component, Input, ViewEncapsulation, ElementRef, Renderer2, } from '@angular/core';
export class CarouselItemComponent {
    constructor(renderer, elRef) {
        this.renderer = renderer;
        this.elRef = elRef;
    }
    getStyles() {
        const slideIndex = this.activeItem - this.itemId;
        if (this.slide) {
            if (slideIndex < 0) {
                return {
                    position: 'absolute',
                    left: '100%',
                };
            }
            else if (slideIndex > 0) {
                return {
                    position: 'absolute',
                    left: '-100%',
                };
            }
            else {
                return {
                    left: '0',
                };
            }
        }
        else {
            return {
                left: '0',
            };
        }
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
}
CarouselItemComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBCarousel-item',
                template: "<div #carouselItem data-test=\"carousel-item\" [class.carousel-slide-item]=\"slide\" class=\"carousel-item {{class}}\"\n    [style]='style' [class.active]='true' [ngStyle]=\"getStyles()\">\n    <ng-content></ng-content>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
CarouselItemComponent.ctorParameters = () => [
    { type: Renderer2 },
    { type: ElementRef }
];
CarouselItemComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    itemId: [{ type: Input }],
    activeItem: [{ type: Input }],
    slide: [{ type: Input }]
};
