// ng-cdbangular/esm2015/lib/components/table/directives/cdb-table.directive.js
import { Component, ElementRef, HostBinding, Input, Renderer2, ViewEncapsulation } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
export class CdbTableDirective {
    constructor(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.entries = 5;
        this._activePage = new BehaviorSubject(0);
        this._dataSource = [];
        this._entries = 5;
        this.pages = [];
        this.paginateRows = () => {
            let pages = [];
            const pagesAmount = this.numberOfPages();
            for (let i = 1; i <= pagesAmount; i++) {
                const pageEndIndex = i * this.getEntries();
                pages.push(this.getDataSource().slice(pageEndIndex - this.getEntries(), pageEndIndex));
            }
            this.pages = pages;
        };
    }
    setDataSource(data) {
        this._dataSource = data;
    }
    getDataSource() {
        return this._dataSource;
    }
    setEntries(num) {
        this._entries = num;
    }
    getEntries() {
        return this._entries;
    }
    getPages() {
        return this.pages;
    }
    setPages() {
        this.paginateRows();
    }
    changeActivePage(page) {
        this._activePage.next(page);
    }
    getActivePage() {
        return this.activePage;
    }
    filterData(searchKey) {
        this.changeActivePage(0);
        return this.getDataSource().filter((data) => {
            return Object.keys(data).some((key) => {
                if (data[key]) {
                    return JSON.stringify(data)
                        .toLowerCase()
                        .includes(searchKey);
                }
            });
        });
    }
    numberOfPages() {
        return Math.ceil(this.getDataSource().length / this.getEntries());
    }
    ngOnInit() {
        this._activePage.subscribe(data => {
            this.activePage = data;
        });
        this.renderer.addClass(this.el.nativeElement, 'table');
        // this.paginateRowsInitialy()
    }
}
CdbTableDirective.decorators = [
    { type: Component, args: [{
                selector: '[CdbTable]',
                exportAs: 'CdbTable',
                template: '<ng-content></ng-content>',
                encapsulation: ViewEncapsulation.None
            },] }
];
CdbTableDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
CdbTableDirective.propDecorators = {
    striped: [{ type: Input }, { type: HostBinding, args: ['class.table-striped',] }],
    autoWidth: [{ type: Input }, { type: HostBinding, args: ['class.w-auto',] }],
    bordered: [{ type: Input }, { type: HostBinding, args: ['class.table-bordered',] }],
    borderless: [{ type: Input }, { type: HostBinding, args: ['class.table-borderless',] }],
    btn: [{ type: Input }, { type: HostBinding, args: ['class.btn-table',] }],
    fixed: [{ type: Input }, { type: HostBinding, args: ['class.table-fixed',] }],
    hoverable: [{ type: Input }, { type: HostBinding, args: ['class.table-hover',] }],
    small: [{ type: Input }, { type: HostBinding, args: ['class.table-sm',] }],
    responsive: [{ type: Input }, { type: HostBinding, args: ['class.table-responsive',] }],
    entries: [{ type: Input }]
};
