// ng-cdbangular/esm2015/lib/components/table/directives/cdb-table-sort.directive.js
import { Directive, ElementRef, EventEmitter, HostListener, Input, Output, Renderer2 } from '@angular/core';
export class CdbTableSortDirective {
    constructor(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.sortEnd = new EventEmitter();
        this.dataSource = [];
        this.direction = 'asc';
    }
    onclick() {
        if (this.enableSort === 'enabled') {
            this.sortData(this.trimWhiteSigns(this.sortBy.toString().toLowerCase()));
        }
    }
    trimWhiteSigns(str) {
        return str.replace(/ /g, '');
    }
    sortData(key) {
        this.dataSource.sort((a, b) => {
            return this.direction === "asc"
                ? a[key] < b[key]
                    ? -1
                    : 1
                : a[key] > b[key]
                    ? -1
                    : 1;
        });
        this.tableEl.setDataSource(this.dataSource);
        this.tableEl.setPages();
        this.sortEnd.emit();
        this.direction = this.direction === 'asc' ? 'dsc' : 'asc';
    }
}
CdbTableSortDirective.decorators = [
    { type: Directive, args: [{
                selector: '[CdbTableSort]'
            },] }
];
CdbTableSortDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
CdbTableSortDirective.propDecorators = {
    sortEnd: [{ type: Output }],
    tableEl: [{ type: Input }],
    dataSource: [{ type: Input, args: ['CdbTableSort',] }],
    sortBy: [{ type: Input }],
    enableSort: [{ type: Input }],
    onclick: [{ type: HostListener, args: ['click',] }]
};
