// ng-cdbangular/esm2015/lib/components/pagination/pageitem/pageitem.component.js
import { Component, ViewEncapsulation, Input, ElementRef, Renderer2, ViewChild, } from '@angular/core';
export class PageitemComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.disabled = false;
        this.active = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
}
PageitemComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBPageitem',
                template: "<li #pageitem class=\"page-item {{class}}\" [style]=\"style\" [class.disabled]=\"disabled\" [class.active]=\"active\" >\n    <ng-content></ng-content>\n</li>",
                encapsulation: ViewEncapsulation.None,
                styles: [".page-item{cursor:pointer;margin:0 7.5px}.page-item a{color:inherit;text-decoration:none}"]
            },] }
];
PageitemComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
PageitemComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    disabled: [{ type: Input }],
    active: [{ type: Input }],
    pageitem: [{ type: ViewChild, args: ['pageitem', { static: false },] }]
};
