// ng-cdbangular/esm2015/lib/components/dropdown/dropdown-item/dropdown-item.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class DropdownItemComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.active = false;
        this.disabled = false;
        this.divider = false;
        this.header = false;
        this.toggle = true;
        this.tabindex = 0;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
        if (this.divider || this.header || this.disabled === true) {
            this.tabindex = -1;
        }
    }
    ngAfterViewInit() {
        if (this.active === true) {
            this.renderer.addClass(this.dropdownItem.nativeElement, 'active');
        }
        if (this.disabled === true) {
            this.renderer.addClass(this.dropdownItem.nativeElement, 'disabled');
        }
        if (this.divider || this.header === false) {
            this.renderer.addClass(this.dropdownItem.nativeElement, 'dropdown-item');
        }
        if (this.divider === true) {
            this.renderer.removeClass(this.dropdownItem.nativeElement, 'dropdown-item');
            this.renderer.addClass(this.dropdownItem.nativeElement, 'dropdown-divider');
        }
        if (this.header === true) {
            this.renderer.removeClass(this.dropdownItem.nativeElement, 'dropdown-item');
            this.renderer.addClass(this.dropdownItem.nativeElement, 'dropdown-header');
        }
    }
}
DropdownItemComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBDropDownItem',
                template: "<div #dropdownItem [tabindex]=\"tabindex\" class=\"{{ class }}\" [style]=\"style\">\n    <ng-content></ng-content>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
DropdownItemComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
DropdownItemComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    active: [{ type: Input }],
    disabled: [{ type: Input }],
    divider: [{ type: Input }],
    header: [{ type: Input }],
    toggle: [{ type: Input }],
    dropdownItem: [{ type: ViewChild, args: ['dropdownItem',] }]
};
