// ng-cdbangular/esm2015/lib/components/badge/badge.component.js
import { Component, Input, ElementRef, Renderer2, ViewEncapsulation, } from '@angular/core';
export class BadgeComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.color = 'primary';
        this.borderType = 'box';
        this.size = 'medium';
        this.flat = false;
    }
    ngOnInit() {
        this.renderer.addClass(this.elRef.nativeElement, 'badge');
        this.renderer.addClass(this.elRef.nativeElement, this.color);
        this.renderer.addClass(this.elRef.nativeElement, this.borderType);
        if (this.size === 'small') {
            this.renderer.addClass(this.elRef.nativeElement, 'CDBBadge-Small');
        }
        else if (this.size === 'medium') {
            this.renderer.addClass(this.elRef.nativeElement, 'CDBBadge-Medium');
        }
        else if (this.size === 'large') {
            this.renderer.addClass(this.elRef.nativeElement, 'CDBBadge-Large');
        }
        if (!this.flat) {
            this.renderer.addClass(this.elRef.nativeElement, 'not-flat');
        }
    }
}
BadgeComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBBadge',
                template: "<span role=\"badge\">\n    <ng-content></ng-content>\n</span>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}.badge{-moz-transition:color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;-ms-transition:color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;-o-transition:color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;-webkit-transition:color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;border-radius:.25rem;display:inline-block;font-size:75%;font-weight:700;font-weight:400;line-height:1;padding:.25em .4em;text-align:center;transition:color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;vertical-align:initial;white-space:nowrap}.box{border-radius:0}.pill{border-radius:20px}.CDBBadge-Large{font-size:1rem;padding:10px 15px}.CDBBadge-Medium{font-size:.65rem;padding:7.5px 8.75px}.CDBBadge-Small{font-size:.5rem;padding:4.5px 6px}.not-flat{box-shadow:0 2px 5px 0 rgba(0,0,0,.16),0 2px 10px 0 rgba(0,0,0,.12)}"]
            },] }
];
BadgeComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
BadgeComponent.propDecorators = {
    color: [{ type: Input }],
    borderType: [{ type: Input }],
    size: [{ type: Input }],
    flat: [{ type: Input }]
};
