// ng-cdbangular/esm2015/lib/components/modal/modal-footer/modal-footer.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class ModalFooterComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() { }
}
ModalFooterComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBModalFooter',
                template: "<div #modalFooter class=\"modal-footer {{ class }}\" [style]=\"style\">\n    <ng-content></ng-content>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
ModalFooterComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
ModalFooterComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    modalFooter: [{ type: ViewChild, args: ['modalFooter', { static: false },] }]
};
