// ng-cdbangular/esm2015/lib/components/accordion/accordionitem/accordionitem.component.js
import { Component, Input, ViewEncapsulation } from '@angular/core';
export class AccordionitemComponent {
    constructor() {
        this.opened = true;
        this.handleAccordionToggle = () => {
            this.opened = !this.opened;
        };
    }
    ngOnInit() { }
}
AccordionitemComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBAccordionitem',
                template: "<div class=\"accordion\" (click)='handleAccordionToggle()' [ngClass]=\"!opened ? 'accordion-item--opened' : ''\">\n    <div class=\"accordion-item__line {{ accordionHeaderBodyClass }}\">\n        <h3 class=\"accordion-item__title {{ accordionHeaderClass }}\">{{data.title}}</h3>\n        <span class=\"accordion-item__icon\">\n        </span>\n    </div>\n    <div class=\"accordion-item__inner {{ accordionBodyClass }}\">\n        <div class=\"accordion-item__content\">\n            <p class=\"accordion-item__paragraph {{ accordionItemClass }}\">{{data.content}}</p>\n        </div>\n    </div>\n</div>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
AccordionitemComponent.ctorParameters = () => [];
AccordionitemComponent.propDecorators = {
    data: [{ type: Input }],
    accordionHeaderBodyClass: [{ type: Input }],
    accordionHeaderClass: [{ type: Input }],
    accordionBodyClass: [{ type: Input }],
    accordionItemClass: [{ type: Input }]
};
