// ng-cdbangular/esm2015/lib/components/sidebar/sidebar-menuitem/sidebar-menuitem.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
import { Router, NavigationEnd } from '@angular/router';
import { filter } from 'rxjs/operators';
import { SidebarService } from '../sidebar.service';
export class SidebarMenuitemComponent {
    constructor(elRef, renderer, sidebarService, router) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.sidebarService = sidebarService;
        this.router = router;
        this.toggled = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
        this.sidebarService.sidebarState.subscribe((state) => {
            this.toggled = state;
            this.setToggled();
        });
        if (!this.link && !this.href) {
            this.link = '/';
        }
    }
    ngAfterViewInit() {
        this.sidebarService.backgroundColor.subscribe((state) => {
            this.backgroundColor = state;
        });
        this.sidebarService.textColor.subscribe((state) => {
            this.textColor = state;
        });
        if (this.link) {
            // Setting Active Link
            this.router.events
                .pipe(filter((e) => e instanceof NavigationEnd))
                .subscribe((navEnd) => {
                if (navEnd.urlAfterRedirects === this.link) {
                    this.setStyle();
                }
                else {
                    this.removeStyle();
                }
            });
        }
    }
    setToggled() {
        if (this.toggled === true) {
            this.renderer.addClass(this.sidebarMenuItem.nativeElement, 'toggled');
            this.renderer.addClass(this.sidebarMenuItemInner.nativeElement, 'toggled');
        }
        else {
            this.renderer.removeClass(this.sidebarMenuItem.nativeElement, 'toggled');
            this.renderer.removeClass(this.sidebarMenuItemInner.nativeElement, 'toggled');
        }
    }
    setStyle() {
        this.renderer.setStyle(this.sidebarMenuItemInner.nativeElement, 'color', this.backgroundColor);
        this.renderer.setStyle(this.sidebarMenuItemInner.nativeElement, 'background', this.textColor);
    }
    removeStyle() {
        this.renderer.removeAttribute(this.sidebarMenuItemInner.nativeElement, 'style');
    }
}
SidebarMenuitemComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBSidebarMenuItem',
                template: "<ng-container *ngIf=\"link\">\n    <a #sidebarMenuLink [routerLink]=\"link\" class=\"sidebarMenuLink\">\n        <ng-container *ngTemplateOutlet=\"menuItemContent\"></ng-container>\n    </a>\n</ng-container>\n\n<ng-container *ngIf=\"href\">\n    <a #sidebarMenuLink [href]=\"href\" target=\"_blank\" rel=\"noopener\" class=\"sidebarMenuLink\">\n        <ng-container *ngTemplateOutlet=\"menuItemContent\"></ng-container>\n    </a>\n</ng-container>\n\n<ng-template #menuItemContent>\n    <li #sidebarMenuItem class=\"InnerMenuLi {{ class }}\" [style]=\"style\">\n        <div #sidebarMenuItemInner class=\"CDBSidebarMenuItem-default\">\n            <ng-container *ngIf=\"icon\">\n                <i class=\"side-icon fa fa-{{ icon }} fa-lg {{ iconClass }}\"></i>\n            </ng-container>\n\n            <span class=\"item-content\">\n                <ng-content></ng-content>\n            </span>\n            <span *ngIf=\"suffix\" class=\"suffix-wrapper\">\n                <ng-container *ngTemplateOutlet=\"suffix\"></ng-container>\n            </span>\n        </div>\n    </li>\n</ng-template>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}.CDBSidebarMenuItem-default{align-items:center;cursor:pointer;display:flex;height:45px;outline:none;padding:8px 2px 8px 15px;position:relative}.CDBSidebarMenuItem-default .side-icon{margin-right:10px;width:30px}.CDBSidebarMenuItem-default.active{background:#fff;color:#000}.CDBSidebarMenuItem-default .item-content{flex-grow:1;flex-shrink:1;overflow:hidden;text-overflow:ellipsis;white-space:nowrap}.CDBSidebarMenuItem-default .suffix-wrapper{margin-right:23px}.CDBSidebarMenuItem-default.toggled{padding-left:13px;padding-right:13px}.CDBSidebarMenuItem-default.toggled .suffix-wrapper{opacity:0}.InnerMenuLi{list-style:none;margin:10px 15px}.InnerMenuLi.toggled{margin-bottom:0;margin-top:0;position:relative}.InnerMenuLi.toggled:before{bottom:0;content:\"\";cursor:pointer;display:inline-block;left:0;position:absolute;right:0;top:0}.sidebarMenuLink,.sidebarMenuLink:active,.sidebarMenuLink:hover{color:inherit;text-decoration:none}"]
            },] }
];
SidebarMenuitemComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 },
    { type: SidebarService },
    { type: Router }
];
SidebarMenuitemComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    icon: [{ type: Input }],
    iconClass: [{ type: Input }],
    suffix: [{ type: Input }],
    link: [{ type: Input }],
    href: [{ type: Input }],
    sidebarMenuItem: [{ type: ViewChild, args: ['sidebarMenuItem', { static: false },] }],
    sidebarMenuItemInner: [{ type: ViewChild, args: ['sidebarMenuItemInner', { static: false },] }]
};
