// ng-cdbangular/esm2015/lib/components/card/card-text/card-text.component.js
import { Component, Input, Renderer2, ElementRef, ViewChild, ViewEncapsulation, } from '@angular/core';
export class CardTextComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.style = '';
        this.small = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.muted === true) {
            this.renderer.addClass(this.cardText.nativeElement, 'text-muted');
        }
    }
}
CardTextComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBCardText',
                template: "<ng-container [ngSwitch]=\"small === false\">\n\n    <ng-container *ngSwitchCase=\"true\">\n        <p #cardText class=\"card-text {{ class }}\" [style]=\"style\">\n            <ng-container *ngTemplateOutlet=\"content\"></ng-container>\n        </p>\n    </ng-container>\n\n    <ng-container *ngSwitchCase=\"false\">\n        <small #cardText class=\"card-text {{ class }}\" [style]=\"style\">\n            <ng-container *ngTemplateOutlet=\"content\"></ng-container>\n        </small>\n    </ng-container>\n\n    <ng-template #content>\n        <ng-content></ng-content>\n    </ng-template>\n\n</ng-container>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
CardTextComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
CardTextComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    muted: [{ type: Input }],
    small: [{ type: Input }],
    cardText: [{ type: ViewChild, args: ['cardText', { static: false },] }]
};
