// ng-cdbangular/esm2015/lib/components/buttontoolbar/buttontoolbar.component.js
import { Component, ElementRef, Input, Renderer2, ViewEncapsulation, } from '@angular/core';
export class ButtontoolbarComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
    }
    ngOnInit() { }
    ngAfterViewInit() {
        this.renderer.addClass(this.elRef.nativeElement, 'btn-toolbar');
        if (this.class) {
            let classes = this.class.split(' ');
            classes.forEach((className) => {
                this.renderer.addClass(this.elRef.nativeElement, className);
            });
        }
        this.renderer.setAttribute(this.elRef.nativeElement, 'style', this.style);
    }
}
ButtontoolbarComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBBtnTb',
                template: "<ng-content></ng-content>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}"]
            },] }
];
ButtontoolbarComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
ButtontoolbarComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }]
};
