// ng-cdbangular/esm2015/lib/components/navbar/nav-item/nav-item.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class NavItemComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.active === true) {
            this.renderer.addClass(this.navbarItem.nativeElement, 'active');
        }
        if (this.text === true) {
            this.renderer.addClass(this.navbarItem.nativeElement, 'navbar-text');
        }
    }
}
NavItemComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBNavItem',
                template: "<li #navbarItem class=\"nav-item {{ class }}\" [style]=\"style\">\n    <ng-content></ng-content>\n</li>",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
NavItemComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NavItemComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    active: [{ type: Input }],
    text: [{ type: Input }],
    navbarItem: [{ type: ViewChild, args: ['navbarItem', { static: false },] }]
};
