// ng-cdbangular/esm2015/lib/components/dropdown/dropdown-toggle/dropdown-toggle.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class DropdownToggleComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.color = 'primary';
        this.caretDropLeft = false;
        this.caretDropRight = false;
        this.caretDropUp = false;
        this.caretDropDown = false;
        this.disabled = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
        if (this.caretDropLeft === true) {
            this.caretMargin = 'mr-3';
        }
        else if (this.caretDropRight === true) {
            this.caretMargin = 'ml-3';
        }
        else if (this.caretDropUp === true) {
            this.caretMargin = 'ml-3';
        }
        else if (this.caretDropDown === true) {
            this.caretMargin = 'ml-3';
        }
    }
    ngAfterViewInit() {
        if (this.color) {
            this.renderer.addClass(this.dropdownToggle.nativeElement, this.color);
        }
        if (this.size === 'lg') {
            this.renderer.addClass(this.dropdownToggle.nativeElement, 'CDBDropDownToggle-lg');
        }
        if (this.size === 'sm') {
            this.renderer.addClass(this.dropdownToggle.nativeElement, 'CDBDropDownToggle-sm');
        }
        if (this.nav === true) {
            this.renderer.addClass(this.dropdownToggle.nativeElement, 'nav-link');
        }
        if (this.caretDropLeft === true) {
            this.renderer.addClass(this.arrowLeft.nativeElement, 'dropleft');
        }
        else if (this.caretDropRight === true) {
            this.renderer.addClass(this.arrowRight.nativeElement, 'dropright');
        }
        else if (this.caretDropUp === true) {
            this.renderer.addClass(this.arrowUp.nativeElement, 'dropup');
        }
    }
    toggleDropdown(event) {
        if (this.disabled || this.nav === true) {
            return;
        }
    }
}
DropdownToggleComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBDropDownToggle',
                template: "<button #dropdownToggle type=\"button\" class=\"CDBDropDownToggle-default {{ class }}\" [style]=\"style\"\n    (click)=\"toggleDropdown($event)\" [disabled]=\"disabled\">\n\n    <span *ngIf=\"caretDropLeft === true\" class=\"{{ caretMargin }} CDB-pointer\" aria-haspopup=\"true\">\n        <span class=\"arrow-wrapper\">\n            <span #arrowLeft class=\"arrow\"></span>\n        </span>\n    </span>\n\n    <ng-content></ng-content>\n\n    <span *ngIf=\"caretDropRight === true\" class=\"{{ caretMargin }} CDB-pointer\" aria-haspopup=\"true\">\n        <span class=\"arrow-wrapper\">\n            <span #arrowRight class=\"arrow\"></span>\n        </span>\n    </span>\n\n    <span *ngIf=\"caretDropUp === true\" class=\"{{ caretMargin }} CDB-pointer\" aria-haspopup=\"true\">\n        <span class=\"arrow-wrapper\">\n            <span #arrowUp class=\"arrow\"></span>\n        </span>\n    </span>\n\n    <span *ngIf=\"caretDropDown === true\" class=\"{{ caretMargin }} CDB-pointer\" aria-haspopup=\"true\">\n        <span class=\"arrow-wrapper\">\n            <span #arrowDown class=\"arrow\"></span>\n        </span>\n    </span>\n</button>",
                encapsulation: ViewEncapsulation.None,
                styles: [".primary{background-color:#276ef1;border-color:#276ef1;box-sizing:border-box;color:#fff}.primary:hover{background-color:rgba(39,110,241,.7411764705882353);color:#fff}.secondary{background-color:#7356bf;border-color:#7356bf;color:#fff}.secondary:hover{background-color:rgba(115,86,191,.7411764705882353);color:#fff}.success{background-color:#05944f;border-color:#05944f;color:#fff}.success:hover{background-color:rgba(5,148,79,.7411764705882353);color:#fff}.danger{background-color:#e11900;border-color:#e11900;color:#fff}.danger:hover{background-color:rgba(225,25,0,.7411764705882353);color:#fff}.warning{background-color:#ffe975;border-color:#ffe975;color:#66512c}.warning:hover{background-color:rgba(255,233,117,.7411764705882353);color:#66512c}.dark{background-color:#000;border-color:#000;color:#fff}.dark:hover{background-color:rgba(0,0,0,.7411764705882353);color:#fff}.light{background-color:#eee;border-color:#eee;color:#000}.light:hover{background-color:hsla(0,0%,93.3%,.7411764705882353);color:#000}.white{background-color:#fff;border-color:#fff;color:#000}.white:hover{background-color:hsla(0,0%,100%,.8352941176470589);color:#000}.info{background-color:#17a2b8;border-color:#17a2b8;color:#fff}.info:hover{background-color:rgba(23,162,184,.7411764705882353);color:#fff}.CDBDropDownToggle-default{-moz-border-radius:0;-ms-border-radius:0;-o-border-radius:0;-webkit-border-radius:0;border:none;border-radius:0;padding:.8rem 2rem}.CDBDropDownToggle-default:focus,.CDBDropDownToggle-default:hover{border:none;outline:none}.CDBDropDownToggle-circle{-moz-border-radius:30px;-ms-border-radius:30px;-o-border-radius:30px;-webkit-border-radius:30px;border-radius:30px}.CDBDropDownToggle-lg{font-size:1.25rem;padding:20px}.CDBDropDownToggle-sm{font-size:.75rem;padding:5px}.CDB-pointer{pointer-events:none!important}.arrow-wrapper{min-height:10px;min-width:10px;position:relative}.arrow-wrapper .arrow{-moz-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);-o-transform:translate(-50%,-50%);-webkit-transform:translate(-50%,-50%);border-color:#fff;border-style:solid;border-width:0 2px 2px 0;display:inline-block;margin-bottom:5px;padding:2.5px;transform:translate(-50%,-50%);transform:rotate(45deg);vertical-align:middle}.dropup{-moz-transform:rotate(-135deg)!important;-ms-transform:rotate(-135deg)!important;-o-transform:rotate(-135deg)!important;-webkit-transform:rotate(-135deg)!important;transform:rotate(-135deg)!important}.dropleft,.dropup{margin-bottom:0!important}.dropleft{-moz-transform:rotate(135deg)!important;-ms-transform:rotate(135deg)!important;-o-transform:rotate(135deg)!important;-webkit-transform:rotate(135deg)!important;transform:rotate(135deg)!important}.dropright{-moz-transform:rotate(-45deg)!important;-ms-transform:rotate(-45deg)!important;-o-transform:rotate(-45deg)!important;-webkit-transform:rotate(-45deg)!important;margin-bottom:0!important;transform:rotate(-45deg)!important}"]
            },] }
];
DropdownToggleComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
DropdownToggleComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    color: [{ type: Input }],
    caretDropLeft: [{ type: Input }],
    caretDropRight: [{ type: Input }],
    caretDropUp: [{ type: Input }],
    caretDropDown: [{ type: Input }],
    size: [{ type: Input }],
    nav: [{ type: Input }],
    circle: [{ type: Input }],
    disabled: [{ type: Input }],
    dropdownToggle: [{ type: ViewChild, args: ['dropdownToggle', { static: false },] }],
    arrowLeft: [{ type: ViewChild, args: ['arrowLeft', { static: false },] }],
    arrowRight: [{ type: ViewChild, args: ['arrowRight', { static: false },] }],
    arrowUp: [{ type: ViewChild, args: ['arrowUp', { static: false },] }],
    arrowDown: [{ type: ViewChild, args: ['arrowDown', { static: false },] }]
};
