// ng-cdbangular/esm2015/lib/components/spinner/spinner.component.js
import { Component, Input, ViewChild, ElementRef, Renderer2, ViewEncapsulation, } from '@angular/core';
export class SpinnerComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.color = '';
        this.size = '';
        this.multicolor = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.class) {
            let classes = this.class.split(' ');
            classes.forEach((className) => {
                this.renderer.addClass(this.circle.nativeElement, className);
            });
        }
        this.renderer.addClass(this.svg.nativeElement, 'spinner');
        if (this.color) {
            this.renderer.addClass(this.circle.nativeElement, this.color);
        }
        if (this.multicolor) {
            this.renderer.addClass(this.circle.nativeElement, 'multicolor');
        }
        let sizeValue = this.size === 'big' ? '100px' : this.size === 'small' ? '30px' : '65px';
        this.renderer.setAttribute(this.svg.nativeElement, 'width', sizeValue);
        this.renderer.setAttribute(this.svg.nativeElement, 'height', sizeValue);
    }
}
SpinnerComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBSpinner',
                template: "<svg class=\"CDBSpinner\" viewBox=\"0 0 66 66\" #svg xmlns=\"http://www.w3.org/2000/svg\">\n    <circle class=\"CDBSpinnerCircle\" [style]='style' #circle fill=\"none\" stroke-width=\"6\" stroke-linecap=\"round\" cx=\"33\" cy=\"33\" r=\"30\">\n    </circle>\n</svg>",
                encapsulation: ViewEncapsulation.None,
                styles: ["@keyframes rotator{0%{transform:rotate(0deg)}to{transform:rotate(1turn)}}@keyframes colors{0%{stroke:#4285f4}25%{stroke:#de3e35}50%{stroke:#f7c223}75%{stroke:#1b9a59}to{stroke:#4285f4}}@keyframes dash{0%{stroke-dashoffset:187}50%{stroke-dashoffset:46.75;transform:rotate(135deg)}to{stroke-dashoffset:187;transform:rotate(450deg)}}.CDBSpinner{animation-duration:1.4s;animation-iteration-count:infinite;animation-name:rotator;stroke-width:5}.CDBSpinnerCircle{animation:dash 1.4s ease-in-out infinite;stroke:#276ef1;stroke-dasharray:187;stroke-dashoffset:0;transform-origin:center}.CDBSpinnerCircle.warning{stroke:#ffe975}.CDBSpinnerCircle.success{stroke:#05944f}.CDBSpinnerCircle.secondary{stroke:#7356bf}.CDBSpinnerCircle.dark{stroke:#000}.CDBSpinnerCircle.danger{stroke:#e11900}.CDBSpinnerCircle.info{stroke:#17a2b8}.CDBSpinnerCircle.multicolor{animation:dash 1.4s ease-in-out infinite,colors 5.6s ease-in-out infinite}"]
            },] }
];
SpinnerComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
SpinnerComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    multicolor: [{ type: Input }],
    class: [{ type: Input }],
    style: [{ type: Input }],
    circle: [{ type: ViewChild, args: ['circle', { static: false },] }],
    svg: [{ type: ViewChild, args: ['svg', { static: false },] }]
};
