// ng-cdbangular/esm2015/lib/components/card/card-body/card-body.component.js
import { Component, ElementRef, Input, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
export class CardBodyComponent {
    constructor(elRef, renderer) {
        this.elRef = elRef;
        this.renderer = renderer;
        this.style = '';
        this.cascade = false;
    }
    ngOnInit() {
        this.renderer.removeAttribute(this.elRef.nativeElement, 'class');
        this.renderer.removeAttribute(this.elRef.nativeElement, 'style');
    }
    ngAfterViewInit() {
        if (this.cascade === true) {
            this.renderer.addClass(this.cardBody.nativeElement, 'card-body-cascad');
        }
    }
}
CardBodyComponent.decorators = [
    { type: Component, args: [{
                selector: 'CDBCardBody',
                template: "<div #cardBody class=\"card-body {{ class }}\" [style]=\"style\">\n    <ng-content></ng-content>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            },] }
];
CardBodyComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
CardBodyComponent.propDecorators = {
    class: [{ type: Input }],
    style: [{ type: Input }],
    cascade: [{ type: Input }],
    cardBody: [{ type: ViewChild, args: ['cardBody', { static: false },] }]
};
